qemu/qemu.go

```diff
@@ -87,32 +100,113 @@ const (
 	VirtioBalloon DeviceDriver = "virtio-balloon"
 
 	//VhostUserSCSI represents a SCSI vhostuser device type.
-	VhostUserSCSI DeviceDriver = "vhost-user-scsi-pci"
+	VhostUserSCSI DeviceDriver = "vhost-user-scsi"
 
 	//VhostUserNet represents a net vhostuser device type.
-	VhostUserNet DeviceDriver = "virtio-net-pci"
+	VhostUserNet DeviceDriver = "virtio-net"
 
 	//VhostUserBlk represents a block vhostuser device type.
-	VhostUserBlk DeviceDriver = "vhost-user-blk-pci"
+	VhostUserBlk DeviceDriver = "vhost-user-blk"
 
 	//VhostUserFS represents a virtio-fs vhostuser device type
-	VhostUserFS DeviceDriver = "vhost-user-fs-pci"
+	VhostUserFS DeviceDriver = "vhost-user-fs"
 
 	// PCIBridgeDriver represents a PCI bridge device type.
 	PCIBridgeDriver DeviceDriver = "pci-bridge"
 
 	// PCIePCIBridgeDriver represents a PCIe to PCI bridge device type.
 	PCIePCIBridgeDriver DeviceDriver = "pcie-pci-bridge"
 
-	// VirtioBlockCCW is the CCW block device driver
-	VirtioBlockCCW DeviceDriver = "virtio-blk-ccw"
+	// VfioPCI is the vfio driver with PCI transport.
+	VfioPCI DeviceDriver = "vfio-pci"
+
+	// VfioCCW is the vfio driver with CCW transport.
+	VfioCCW DeviceDriver = "vfio-pci"
+
+	// VHostVSockPCI is a generic Vsock vhost device with PCI transport.
+	VHostVSockPCI DeviceDriver = "vhost-vsock-pci"
+)
+
+func isDimmSupported(config *Config) bool {
+	switch runtime.GOARCH {
+	case "amd64", "386":
+		if config != nil {
+			switch config.Machine.Type {
+			case Microvm:
+				return false
+			}
```

Comment:
I would separate the Microvm addition/support into a separate commit, making it clear that the first one is only about supporting multiple transports.

Reply:
Yeah, makes sense.